yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -123,20 +130,28 @@ export class Archiver implements L2BlockSource, UnverifiedDataSource, ContractDa
       this.nextL2BlockFromBlock,
       nextExpectedRollupId,
     );
+
+    // create the block number -> block hash mapping to ensure we retrieve the appropriate events
+    const blockHashMapping: { [key: string]: Buffer } = {};
```

Comment:
```suggestion
    const blockHashMapping: { [key: string]: Buffer | undefined } = {};
```
So the compiler forces you to check that a given key is in the mapping before trying to use it